common/param.go

```diff
@@ -0,0 +1,565 @@
+// Copyright 2021 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package common
+
+import (
+	"fmt"
+	"math"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/viper"
+	structpb "google.golang.org/protobuf/types/known/structpb"
+)
+
+func (c *ParamDescription) Apply(value interface{}) error {
```

Comment:
Just a suggestion: This method can be slightly re-organized to first, check for t and then the default is common.

Also internal switch for Int types or UInt types will further make it neat?

WDYT?

Reply:
It would make it more compact but will make the logic harder to flow and harder to extend for further type conversion. It makes more sense to figure out what your target Kind is and then deal with the conversion rather than to switch between value kinds and then convert based on target.